core/src/main/scala/org/apache/spark/storage/ShuffleBlockFetcherIterator.scala

```diff
@@ -1124,4 +1394,54 @@ object ShuffleBlockFetcherIterator {
    */
   private[storage]
   case class DeferFetchRequestResult(fetchRequest: FetchRequest) extends FetchResult
+
+  /**
+   * Result of an un-successful fetch of either of these:
+   * 1) Remote shuffle block chunk.
+   * 2) Local merged block data.
+   *
+   * Instead of treating this as a FailureFetchResult, we ignore this failure
+   * and fallback to fetch the original unmerged blocks.
+   * @param blockId block id
+   * @param address BlockManager that the merged block was attempted to be fetched from
+   * @param size size of the block, used to update bytesInFlight.
+   * @param isNetworkReqDone Is this the last network request for this host in this fetch
+   *                         request. Used to update reqsInFlight.
+   */
+  private[storage] case class IgnoreFetchResult(blockId: BlockId,
```

Comment:
We are not ignoring the result as such, but using it to initiate a fallback. 
`IgnoreFetchResult` -> `RetriableMergeFailureResult` ? Or something better ?

Reply:
Will `Retriable` be confusing that this request itself is retried? Can call it `FallbackOnFailureFetchResult`. Let me know what you think?

Reply:
Stick Merge in the class name - just in case we have other forms of Fallback in future which need specialized handling.

Reply:
Made this change.